SCons/Platform/__init__.py

```diff
@@ -279,7 +283,8 @@ def tmpfile_cleanup(file) -> None:
         tempfile_esc_func = env.get('TEMPFILEARGESCFUNC', SCons.Subst.quote_spaces)
         args = [tempfile_esc_func(arg) for arg in cmd[1:]]
         join_char = env.get('TEMPFILEARGJOIN', ' ')
-        os.write(fd, bytearray(join_char.join(args) + "\n", encoding="utf-8"))
+        encoding = env.get('TEMPFILEENCODING', 'utf-8')
+        os.write(fd, bytes(join_char.join(args) + "\n", encoding=encoding))
```

Comment:
We're not going to change this - just immediately write it out  - so there's no reason for it to be a bytearray.